src/goto-symex/ssa_step.h

```diff
@@ -0,0 +1,203 @@
+/*******************************************************************\
+
+Module: Symbolic Execution
+
+Author: Romain Brenguier <romain.brenguier@diffblue.com>
+
+*******************************************************************/
+
+#ifndef CPROVER_GOTO_SYMEX_SSA_STEP_H
+#define CPROVER_GOTO_SYMEX_SSA_STEP_H
+
+#include <goto-programs/goto_trace.h>
+#include <solvers/prop/literal.h>
+
+#include "symex_target.h"
+
+/// Single SSA step in the equation. Its `type` is defined as
+/// goto_trace_stept::typet. Every SSA step has a `source` to identify its
+/// origin in the input GOTO program and a `guard` expression which holds
+/// the path condition required to reach this step: they limit the scope of
+/// this step.
+///
+/// SSA steps that represent assignments and declarations also store the
+/// left- and right-hand sides of the assignment. The left-hand side
+/// `ssa_lhs` is required to be of type ssa_exprt: in SSA form, variables
+/// are only assigned once, see \ref static-single-assignment. To achieve
+/// that, we annotate the original name with 3 types of levels, see
+/// ssa_exprt. The assignment step also represents the left-hand side in two
+/// other _full_ forms: `ssa_full_lhs` and `original_full_lhs`, which store
+/// the original expressions from the input GOTO program before removing
+/// array indexes, pointers, etc. The `ssa_full_lhs` uses the level-annotated
+/// names.
+///
+/// Assumptions, assertions, goto steps, and constraints have `cond_expr`
+/// which represent the condition guarding this step, i.e. what must hold for
+/// this step to be taken. Both `guard` and `cond_expr` will later be
+/// translated into verification condition for the SAT/SMT solver (or some
+/// other decision procedure), to be referred by their respective literals.
+/// Constraints usually arise from external conditions, such as memory models
+/// or partial orders: they represent assumptions with global effect.
+///
+/// Function calls store `called_function` name as well as a vector of
+/// arguments `ssa_function_arguments`. The `converted` version of a
+/// variable will contain its version for the SAT/SMT conversion.
+class SSA_stept
+{
+public:
+  symex_targett::sourcet source;
+  goto_trace_stept::typet type;
+
+  // NOLINTNEXTLINE(whitespace/line_length)
```

Comment:
This code has been reformatted, and thus these linter exclusions are no longer appropriate. Please remove all of them.

Reply:
done. I could also reformat `symex_target_equation.h` before moving the code if you think that would make it easier to review

Reply:
Ah, no, don't bother, I wasn't worried about that. I was just surprised seeing those linter exclusions where they seemingly didn't make sense, and then figured out that they were only necessary under the previous formatting.